module/utils.mjs

```diff
@@ -208,3 +209,23 @@ function _localizeObject(obj, keys) {
     }
   }
 }
+
+/**
+ * For unidentified items, returns a masked name based on the item type and base item, where appropriate.
+ * @param {object} item   The item to evaluate.
+ * @returns {string}      The masked name to show for the unidentified item
+ * @private
+ */
+function unidentifiedName(item) {
+  // TODO: use item.system.baseItem || item.system.weaponType / item.system.armor.type when baseItem keys are available
```

Comment:
```suggestion
  const baseItemName = ProficiencySelector.getBaseItem(item.system.baseItem, { indexOnly: true })?.name;
```
Passing `indexOnly` to `getBaseItem` will always return the index with the name synchronously.

Reply:
Oh I missed that flag, thanks!